### ISSUE TITLE
[Bug fix with an enhancement] Standardize "Shop Now" Button Behavior and Link Products to Shops

### ISSUE THREAD

**Issue by kavya410004** [mdazfar2/Ezyshop#222]
I've noticed a few inconsistencies with the "Shop Now" button behavior, "Products" button behavior and how products are linked with shops:

**Current Behavior:**
1. The "Shop Now" button on category cards on the home page redirects to the products page for that category whereas the "Shop Now" button on the category page redirects to the shops page related to that category. 
2. When clicking the "Products" button on the shops page, it currently displays **products related to the category** instead of **products from that specific shop**.

**Proposed Changes:**

Link Products with Shops: _[Enhancement]_
- I will add a shop property to each product so they can be associated with a specific shop.
- Display the shop name alongside other details in the product card for better clarity.

Standardize "Shop Now" Button Behavior: _[Bug Fix]_
- Ensure that the "Shop Now" button on both the home page and category page behaves consistently.
- Both buttons should either take users to the category products page (Let me know if the buttons have to take the user to display relevant shops instead of products). 

Fix Products Display in Shops Page: _[Enhancement]_
- Update the "Products" button on the shops page to display products from the selected shop rather than the entire category-based products.

**Screenshots:**
Products (with no shop name)
![image](https://github.com/user-attachments/assets/66f6f3c6-32c3-4ee3-b98c-71b3a0b8cdf1)

I would like to work on this issue, please assign this to me.
- [x] I have read the Contributing Guidelines
- [x] I'm a GSSOC'24-Extd contributor
- [x] I'm a Hacktoberfest contributor
- [x] I have starred the repository
- [x] I am willing to work on this issue (optional)

**Comment by mdazfar2** [comment]
@kavya410004 Issue acknowledged! You are going to be assigned.

**Comment by mdazfar2** [comment]
Hey @kavya410004 can you do one more thing;

- You created the main feature for the product page, so when any user clicks on the page to be redirected, an alert message should appear saying "please login first." Once the user logs in, they can be redirected to the product or shop page. Please let me know if you're interested in working on this.

**Comment by kavya410004** [comment]
@mdazfar2 Let me rephrase it. 
If a user hasn't logged in and tries to visit products page or shops page, an alert should appear saying "please login first". If the user has already logged in, only then the user would be redirected to products or shops page.  Is this right?

**Comment by mdazfar2** [comment]
Yep, @kavya410004, here I will send you the demo video of the alert message, how it appears and removes itself automatically in about 2-3 seconds:

https://github.com/user-attachments/assets/0c459506-21dc-4dd1-ab87-c7ab674f1779

**Comment by kavya410004** [comment]
Okay @mdazfar2, I can work on this issue. 
In the video the alert message was displayed without redirecting the user to login/ signup page. Is displaying the message alone is enough or is redirection to login page also required?

**Comment by mdazfar2** [comment]
@kavya410004 When we redirect to the login/signup page, this functionality makes the user irritated, so we can add only the alert message without redirecting to the login page.

**Comment by kavya410004** [comment]
@mdazfar2 , I have raised the issue here #225 . 
I will be working on issue #222 first and move on to the next issue #255 
Thank you.

**Comment by kavya410004** [comment]
Hey @mdazfar2 ! Just noticed my PR got merged yesterday, but the issue is still open. Did you miss closing it? Thanks!

**Comment by mdazfar2** [comment]
Thanks for letting me know @kavya410004
